frontend/src/pages/Compare.tsx

```diff
@@ -14,56 +14,115 @@
+  const [compareVersion, setCompareVersion] = useState<CompareVersion>(CompareVersion.Unknown);
   const queryParamRunIds = new URLParser(props).get(QUERY_PARAMS.runlist);
   const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
 
-  // Retrieves run details.
-  const { isSuccess, isError, data } = useQuery<ApiRunDetail[], Error>(
+  // Retrieves run details, set page version on success.
+  const { isError, error } = useQuery<ApiRunDetail[], Error>(
     ['run_details', { ids: runIds }],
     () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
     {
       staleTime: Infinity,
-      onError: async error => {
+      onSuccess: data => {
+        // Set the version based on the runs included.
+        if (data.length < 2 || data.length > 10) {
+          setCompareVersion(CompareVersion.InvalidRunCount);
+        } else {
+          const v2runs = data.filter(run =>
+            run.run?.pipeline_spec?.hasOwnProperty('pipeline_manifest'),
+          );
+          if (v2runs.length === 0) {
+            setCompareVersion(CompareVersion.V1);
+          } else if (v2runs.length === data.length) {
+            setCompareVersion(CompareVersion.V2);
+          } else {
+            setCompareVersion(CompareVersion.Mixed);
+          }
+        }
+      },
+    },
+  );
+
+  useEffect(() => {
+    // Update banner based on error, feature flag, run versions, and run count.
+    if (isError) {
+      (async function() {
         const errorMessage = await errorToMessage(error);
-        props.updateBanner({
+        updateBanner({
           additionalInfo: errorMessage ? errorMessage : undefined,
           message: `Error: failed loading ${runIds.length} runs. Click Details for more information.`,
           mode: 'error',
         });
-      },
-      onSuccess: () => props.updateBanner({}),
-    },
-  );
+      })();
+    } else if (
+      isFeatureEnabled(FeatureKey.V2_ALPHA) &&
+      compareVersion === CompareVersion.InvalidRunCount
+    ) {
+      updateBanner({
+        additionalInfo:
+          'At least two runs and at most ten runs must be selected to view the Run Comparison page.',
+        message:
+          'Error: failed loading the Run Comparison page. Click Details for more information.',
+        mode: 'error',
+      });
+    } else if (isFeatureEnabled(FeatureKey.V2_ALPHA) && compareVersion === CompareVersion.Mixed) {
+      updateBanner({
+        additionalInfo:
+          'The selected runs are a mix of V1 and V2.' +
+          ' Please select all V1 or all V2 runs to view the associated Run Comparison page.',
+        message:
+          'Error: failed loading the Run Comparison page. Click Details for more information.',
+        mode: 'error',
+      });
+    } else if (isFeatureEnabled(FeatureKey.V2_ALPHA) && compareVersion !== CompareVersion.V1) {
+      // Clear the banner unless the V1 page is shown, as that page handles its own banner state.
+      updateBanner({});
+    }
```

Comment:
I only clear the banner here if the V1 page is not going to be shown. If this check is removed, it can overwrite the banner on the V1 page.

Reply:
Resolving this comment as it has been reviewed.